### PR TITLE
feat/669 highlight lightbox

### LINKED ISSUE #658: Header should be sticky as user scrolls down

### LINKED ISSUE #669: Put highlight threads in a lightbox
The highlight threads are cramped on the side, and the UX on mobile doesn't work. Bad design decision by me.

When user highlights or clicks on an existing highlight, it should reveal the option to tweet or comment the same way it does now; if they click the little comment icon, it should open a lightbox that takes up most of the screen; the lightbox should be a bigger experience of the comment threading that is currently off to the right side. This should also work on mobile.

Also, the highlights appear on the right side of the page without the comments below them, and clicking on one of them opens the lightbox discussion.

### PR DESCRIPTION
**Pull request by sterlingrules**
Fixes #658 
Fixes #669

### CODE REVIEW

File: src/components/Highlightable.vue [hunk 1]
```diff
@@ -248,20 +254,33 @@ export default {
...
     },
 
     onReply () {
-      this.isReplying = true
+      const query = {
+        thread_id: this.entityId,
+        comment_id: this.parentCommentId,
+      }
+
+      this.$router.push({ query }).catch((err) => {})
```
Comment (marcosschlup) [inline src/components/Highlightable.vue]: I think you don't need `catch`

File: src/components/Highlightable.vue [hunk 2]
```diff
@@ -248,20 +254,33 @@ export default {
...
         return this.onReply()
       }
 
-      this.isHighlighting = true
+      const query = {
+        thread_id: this.entityId,
+        comment_id: undefined,
+      }
+
+      this.$router.push({ query }).catch((err) => {})
```
Comment (marcosschlup) [inline src/components/Highlightable.vue]: same as above?

File: src/components/comment/CommentsList.vue [hunk 3]
```diff
@@ -104,6 +118,21 @@ export default {
...
+      if (!this.preview) {
+        return
+      }
+
+      const query = {
+        thread_id: this.forumThreadId,
+        comment_id: evt.target.getAttribute('id'),
+      }
+
+      this.$router.push({ query }).catch((err) => {})
```
Comment (marcosschlup) [inline src/components/comment/CommentsList.vue]: Same `catch`

File: src/components/post/PostHighlights.vue [hunk 4]
```diff
@@ -0,0 +1,240 @@
+<template>
+  <div class="modal" :class="{ 'is-active': isActive }">
+    <div class="modal-content">
+      <div class="modal-header">
+        <strong class="title">{{ commentCount || 0 }}  {{commentCount == 1 ? 'highlight' : 'highlights'}}</strong>
```
Comment (marcosschlup) [inline src/components/post/PostHighlights.vue]: I think default for `commentCount` is already `0`

File: src/components/post/PostHighlights.vue [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+  data () {
+    return {
+      isActive: false,
+      commentId: null,
+      threadId: null,
+    }
+  },
+
+  watch: {
+    $route (to, from) {
```
Comment (marcosschlup) [inline src/components/post/PostHighlights.vue]: It looks like there's another option for this `watch` case:
https://router.vuejs.org/guide/advanced/navigation-guards.html#in-component-guards
I don't know if it works in our app

File: src/components/post/PostHighlights.vue [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+  },
+
+  computed: {
+    canComment () {
+      return (!this.comments.filter(c => c.highlight === this.highlight).length)
+    },
+
+    commentsSorted () {
+      return clone(this.comments).sort((a, b) => {
+        return new Date(b.dateCreated) - new Date(a.dateCreated)
```
Comment (marcosschlup) [inline src/components/post/PostHighlights.vue]: Is this date really a `string`? For me, in something similar, this just works:

```javascript
return o1.dateCreated >= o2.dateCreated ? -1 : 1;
```

File: src/components/post/PostHighlights.vue [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+      if (this.$router.currentRoute.query.thread_id ||
+          this.$router.currentRoute.query.comment_id) {
+        this.$router.push({ query })
+      }
+    },
+  },
+}
+</script>
+
+<style lang="scss">
```
Comment (marcosschlup) [inline src/components/post/PostHighlights.vue]: I thought the application default was `stylus`

File: src/components/post/PostHighlights.vue (hunk 6 above)
Comment (sterlingrules) [reply src/components/post/PostHighlights.vue]: Yes, this date is a string.

File: src/components/post/PostHighlights.vue (hunk 5 above)
Comment (sterlingrules) [reply src/components/post/PostHighlights.vue]: Does this option have a benefit over how it's currently implemented?

**Pull request merged**